nx-X11/programs/Xserver/hw/nxagent/Clipboard.c

```diff
@@ -921,15 +929,34 @@ void nxagentHandleSelectionClearFromXServerByIndex(int index)
      * Set the root window with the NullClient as selection owner. Our
      * clients asking for the owner via XGetSelectionOwner() will get
      * this for an answer.
+     * Set the CurrentSelection data just as ProcSetSelectionOwner does.
      */
+    CurrentSelections[index].lastTimeChanged = time;
     CurrentSelections[index].window = screenInfo.screens[0]->root->drawable.id;
+    CurrentSelections[index].pWin = NULL;
     CurrentSelections[index].client = NullClient;
 
     clearSelectionOwnerData(index);
 
     setClientSelectionStage(index, SelectionStageNone);
 
     invalidateTargetCache(index);
+
+    /*
+     * now call the callbacks. This is important, especially for
+     * xfixes events to work properly. Keep in mind that this will
+     * also call out own callback so we must be prepared there to not
```

Comment:
s/own callback/own callbacks/ ?

Reply:
or s/out/our/ ?

Reply:
no, we only have one callback. Of course it must read "our"...